build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/JavaModulePrecommitTask.java

```diff
@@ -116,8 +116,11 @@ private void checkModuleVersion(ModuleReference mref) {
 
     private void checkModuleNamePrefix(ModuleReference mref) {
         getLogger().info("{} checking module name prefix for {}", this, mref.descriptor().name());
-        if (mref.descriptor().name().startsWith("org.elasticsearch.") == false) {
-            throw new GradleException("Expected name starting with \"org.elasticsearch.\", in " + mref.descriptor());
+        if (mref.descriptor().name().startsWith("org.elasticsearch.") == false
```

Comment:
Rather than hardcoding this, I wonder if it is something serverless should pass in (say via a flag or some other Gradle mechanism for composite builds), so that we don't start allowing co.elasticsearch to leak into the main repo. But I leave this up to @mark-vieira.

Reply:
Yeah, I think we should make this task more extensible. This same task also enforces module version, which might not always be applicable. If this is blocking other work we can probably disable this in serverless for now. The only check I think would be worthwhile keeping there is the services bit.

Reply:
I am creating a new module in serverless where I wanted to use a name of a module with `co.elasticsearch.etc.etc..`
if we can disable it, it would work for me